network_simulator/plot_tool.py

```diff
@@ -0,0 +1,390 @@
+                "Packet Sent", "(pkts / sec)",
+                flow_stats.packet_sent_times)
+
+            # Packet receive times.
+            plt.subplot(435)
+            rec_rate_lst = PlotTool().gen_rate_tuple_list(
+                flow_stats.packet_rec_times)
+            PlotTool().graph_tuple_list(rec_rate_lst)
+            receive_legend.append(flow_id)
+            # Output average packet received per sec to stdout.
+            PlotTool().output_count_avg_tuple_list(
+                "Flow " + flow_id,
+                "Packet Received", "(pkts / sec)",
+                flow_stats.packet_rec_times)
+
+            # Packet RTT times.
+            plt.subplot(438)
+            PlotTool().graph_tuple_list(flow_stats.packet_rtts)
+            rtt_legend.append(flow_id)
+            # Output average packet RTT to stdout.
+            PlotTool().output_sum_avg_tuple_list(
+                "Flow " + flow_id,
+                "Packet RTT", True, "(sec)",
+                flow_stats.packet_rtts)
+
+            # Window size in packets w.r.t time.
+            plt.subplot(4,3,11)
+            PlotTool().graph_tuple_list(flow_stats.window_size_times)
+            window_legend.append(flow_id)
+
+        # Finalize sent graph.
+        plt.subplot(432)
+        plt.legend(sent_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Flow send rate (Mbps)")
+
+        # Finalize receive graph.
+        plt.subplot(435)
+        plt.legend(receive_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Flow Receive Rate (Mbps)")
+
+        # Finalize RTT graph.
+        plt.subplot(438)
+        plt.legend(rtt_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("RTT (sec)")
+
+        # Finalize window size graph.
+        plt.subplot(4,3,11)
+        plt.legend(window_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Window Size (pkts)")
+
+    def graph_hosts(self):
+        """
+        Graphs the host stats of each available host.
+
+        This includes host send and receive rates.
+        """
+        # Set up legend storage for host names, which are their addresses
+        # in strings.
+        # Packet sent rate legend.
+        sent_legend = []
+        # Packet received rate legend.
+        receive_legend = []
+
+        # Iterate through each host to display each stats.
+        for host_addr, host_stats in self.stats.host_stats.iteritems():
+            # Name of host would be "address".
+            # Packet sent times.
+            plt.subplot(433)
+            sent_rate_lst = PlotTool().gen_rate_tuple_list(
+                host_stats.packet_sent_times)
+            PlotTool().graph_tuple_list(sent_rate_lst)
+            sent_legend.append(host_addr)
+            # Output average packet received per sec to stdout.
+            PlotTool().output_count_avg_tuple_list(
```

Comment:
Same as above; need to output the average send rate (for this host) in Mbps. Also make sure to update the comments.